### PR TITLE
first installment of the wiki page

### PR DESCRIPTION
**Pull request by Walrussuit101**
See issue #36 for features (been keeping a checklist there)

Going forward I'm going to split up the tasks into their own branches. Sorry about this massive PR. I should've known how generalized "feature/wikiCRUD" was :/ Not going to link issue as it shouldn't be closed when this merges.

Change Shortlist:
- Still using fetch().then promises, Tom and I talked about having to look at this after the milestone (to use async/wait)
- Create/read wikis
- Create/read/update/delete entries
- Create/read comments
- Create/read headings
- Read sidebars
- Added ```/wikinotfound``` route that users will be routed too if they somehow choose a non-valid ```wiki/{id}``` route

### CODE REVIEW

File: src/components/Wiki/EntryEditForm/EntryEditForm.tsx [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+                this.setState({
+                    entryData: this.props.entryData,
+                    sideBarData: this.props.sideBarData,
+                    sideBarElements: [],
+                    headingEditElements: this.props.headingEditElements
+                })
+                this.buildSideBarElements();
+            }, 200)
+        }
+        // else if(prevState.sideBarData.content !== this.state.sideBarData.content){
```
Comment (Rithonn) [inline src/components/Wiki/EntryEditForm/EntryEditForm.tsx:156]: Remove commented-out code if it is not going to be used.

**Review by Rithonn — CHANGES_REQUESTED** [review]
Overall good stuff. There will be some items that will need to be cleaned later on, but after you remove the commented out code this should be good to be merged.

**Review by cbrandt6 — APPROVED** [review]
Good stuff here Tim

**Comment by Rithonn** [comment]
Might want to have it down the road when a new wiki is created have the user be redirected to the newly created wiki

File: src/components/Wiki/EntryEditForm/EntryEditForm.tsx (hunk 1 above)
Comment (Walrussuit101) [reply src/components/Wiki/EntryEditForm/EntryEditForm.tsx:156]: Going to use that for updating headings, so it will be used :thumbsup:

**Comment by Walrussuit101** [comment]
> Might want to have it down the road when a new wiki is created have the user be redirected to the newly created wiki

[483e131](https://github.com/YCP-Swarm-Robotics-Capstone-2020-2021/swarm-website-frontend/pull/53/commits/483e131572d9a0fb38b77f003d1899a396626106)

**Review by Rithonn — APPROVED** [review]
Good Stuff

**Pull request merged**